packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -148,14 +180,33 @@ class MediaQueryData {
   /// for subsequent descendants in the widget tree by inserting a new
   /// [MediaQuery] widget using the [MediaQuery.removePadding] factory.
   ///
+  /// Padding is derived from the values of viewInsets and viewPadding.
+  ///
   /// See also:
   ///
-  ///  * [MediaQueryData], which provides some additional detail about this
-  ///    property and how it differs from [viewInsets].
+  ///  * [ui.window], which provides some additional detail about this
+  ///    property and how it relates to [viewInsets] and [viewPadding].
   ///  * [SafeArea], a widget that consumes this padding with a [Padding] widget
   ///    and automatically removes it from the [MediaQuery] for its child.
   final EdgeInsets padding;
 
+  /// The parts of the display that may be obscured by the system UI, but
+  /// respect the padding established below the obstruction.
+  ///
+  /// When a mobile device's keyboard is visible, `viewPadding.bottom`
+  /// corresponds to the bottom of the view beneath it.
+  ///
+  /// This value is independent of the [padding] and [viewInsets]: their values
+  /// are measured from the edges of the [MediaQuery] widget's bounds. The
+  /// bounds of the top level MediaQuery created by [WidgetsApp] are the
+  /// same as the window (often the mobile device screen) that contains the app.
```

Comment:
I think we can remove the parnethetical remark.  Maybe we could add another sentence saying `On mobile devices, this will typically be the full screen.`?